### ISSUE TITLE
Post method with zip file

### ISSUE THREAD

**Issue by rapw3k** [swagger-api/swagger-editor#1017]
**Swagger File**

``` yaml
swagger: '2.0'
info:
  version: 1.0.0
  title: Basic Example
  description: |
    An example 
host: sandbox.wf4ever-project.org
basePath: /rodl
consumes:
  - application/zip
  - multipart/form-data
produces:
  - text/plain
  - text/turtle
schemes:
  - http
  - https
securityDefinitions:
  Bearer:
    type: apiKey
    name: Authorization
    in: header
paths:
  /ROs:
    get:
      security:
        - Bearer: []
      responses:
        '200':
          description: >-
            Will send `Authenticated` if authentication is succesful, otherwise
            it will send `Unauthorised`
  /zip/create:
    post:
      security:
        - Bearer: []
      summary: Creates a new Reserch Object
      description: Creates a new Reserch Object
      responses:
        '201':
          description: Created
      parameters:
        - name: slug
          in: header
          description: Identifier of the research object.
          type: string
        - name: zipfile
          in: formData
          description: content in zip format
          type: file

```
- Version: 2.0 (running on a linux ubuntu server)
- Browser/OS: Chrome and Safari (with cross-origin resource sharing disabled), Mac OSX El Capitan

**Issue**

Hi, 
I am trying to use the method "/zip/create" above, but I am unable to send correctly the request. After reading documentation, I understand in order to post a file you need to send it using "formData" and type "file", and use "multipart/form-data" in the consumes parameter. However, this returns: Status Code:415 Unsupported Media Type (see screenshot below). I cannot find the way to specify the "application/zip" as well in the call. How is this possible?

Thanks in advance.
Raul

You can test using the authentication: Bearer 4e5b6138-5885-4c8f-a794-cf368c67a4e4
<img width="1278" alt="screen shot 2016-07-22 at 17 21 34" src="https://cloud.githubusercontent.com/assets/557493/17062012/77306ae8-5031-11e6-8d08-2f7855581373.png">

**Comment by saharj** [comment]
Tryoperation has some bugs, specially with file uploads. Maybe we should refactor it.

**Comment by rapw3k** [comment]
Thanks, what do you mean by try operation? How to track any progress on this?

**Comment by saharj** [comment]
Try Operation is the component you see after you click on `Try this operation`.

**Comment by rapw3k** [comment]
Hi,
I think my issue here is exactly the problem described in [https://github.com/swagger-api/swagger-core/issues/636](url) and [https://github.com/OAI/OpenAPI-Specification/issues/222](url). 

@webron mentioned that this feature request was added for the next proposal (Dec 2, 2014). Is there any news on this? 

Alternatively @webron also mentioned that vendor extensions can be used to extend this functionality. Could you point to those vendor extensions?

**Comment by webron** [comment]
There's no solution for it yet, and we haven't covered that in the talks about the next version of the spec. Vendor extensions are pretty much *your* extensions that you write, they will not add functionality magically, but they can be used by various tools to extend the functionality. If you want a tool to understand your extension, it's up to you to modify it to support it.

**Comment by rapw3k** [comment]
Alternatively, @webron is there any way to send the file in the body, like the equivalent in curl:

curl -X POST  --verbose --data-binary "@aggregatedNoFolder.zip" -H "slug: mytest2" -H "Accept: application/json" -H "Content-Type: application/zip" -H "Authorization: Bearer 4e5b6138-5885-4c8f-a794-cf368c67a4e4" http://sandbox.wf4ever-project.org/rodl/zip/create

**Comment by webron** [comment]
Sure, the tooling may not quite support it, but you can always say the method `consumes` `application/zip` and has a single empty schema as the body (since there's no real way to represent it).

**Comment by rapw3k** [comment]
Thanks for the answer, but I am not sure how would you do something like this in the swagger file. I can say that the file parameter is in body, but how to send it ? how would you define the parameter to pass the zip file to the service (e.g., zipfile parameter in the example below) @webron ?

''' yaml
parameters:
        - name: slug
          in: header
          description: Identifier of the research object.
          type: string
        - name: zipfile
          in: formData
          description: content in zip format
          type: file


something like this won't pass any file:

        -name: zipfile
          in: body
          description: content in zip format
          schema:
            type: string

**Comment by webron** [comment]
I don't really understand the question.

**Comment by rapw3k** [comment]
The question is how can you send a zip file in a POST method in the swagger file, to achieve the same behaviour as the curl command below:

curl -X POST --verbose --data-binary "@aggregatedNoFolder.zip" -H "slug: mytest2" -H "Accept: application/json" -H "Content-Type: application/zip" -H "Authorization: Bearer 4e5b6138-5885-4c8f-a794-cf368c67a4e4" http://sandbox.wf4ever-project.org/rodl/zip/create

At the moment I can send the file using parameter type "formData", but I cannot say that its a zip file, as the consumes has to be "multipart/form-data", so I receive Status Code:415 Unsupported Media Type. With a parameter type "body", as you proposed, I don't see how can I send a file.

**Comment by webron** [comment]
Okay, let's separate things. Do what you want to do, you'd need to set it as a body parameter. The *tools*, specifically, the editor and the ui, do not support this form of file upload, and I doubt they will soon.

**Comment by saharj** [comment]
[This PR](https://github.com/swagger-api/swagger-editor/pull/1070) may has fixed this issue.

**Comment by saharj** [comment]
This issue should be fixed with the latest PR #1141 . Please let us know if you still see the issue.
